### DISCUSSION TITLE
How can I provide/configure my own implementation of SessionStore

### DISCUSSION THREAD

**Discussion by organic-scholar** [micronaut-projects/micronaut-core#4990] [Q&A] [answered]
Micronaut provides an InMemory and Redis-based session storage. I want to provide a custom storage implementation. I wasn't able to find something useful in guides/docs.

Please provide some insight is it possible and how.

**Comment by graemerocher** [comment, marked as answer]
It is just a matter of implementing the `io.micronaut.session.SessionStore` interface and defining the implementation as a bean. Here is another example written for Coherence https://github.com/micronaut-projects/micronaut-coherence/blob/master/coherence-session/src/main/java/io/micronaut/coherence/httpsession/CoherenceSessionStore.java

**Reply by organic-scholar** [reply]
Thanks @graemerocher it is helpful
